### DISCUSSION TITLE
Update widget based on form input

### DISCUSSION THREAD

**Discussion by Konafets** [filamentphp/filament#5443] [Help] [answered]
Hi, 

I want to add a widget on the creation page and want to updated it with information from the form interactively. Some of the data is the literal input from the form and other is calculated data from multiple input fields. Is that possible?

**Comment by danharrin** [comment]
If you know Livewire, it's possible. You need to emit events and listen for them between the page and the widget.

**Comment by Konafets** [comment, marked as answer]
That was a hint in the right direction. I emitted an event from within a closure and catched it inside the widget:

```php
Select::make('federal_state_id')
// ...
->afterStateUpdated(function ($livewire, int $state) {
    $realEstateTransferTax = 0.0;
    $federalState = FederalState::find($state);

    if ($federalState instanceof FederalState) {
        $realEstateTransferTax = $federalState->real_estate_transfer_tax;
    }

    $livewire->emit('updateLandRegistryTax', $realEstateTransferTax);
 }),

 ```
